report_qweb_pdf_watermark/models/report.py

```diff
@@ -108,11 +111,29 @@ def _run_wkhtmltopdf(
             return result
 
         pdf = PdfFileWriter()
-        pdf_watermark = None
+
         try:
             pdf_watermark = PdfFileReader(BytesIO(watermark))
         except PdfReadError:
-            # let's see if we can convert this with pillow
+            pdf_watermark = None
+        except UnicodeDecodeError as e:
+            # workaround: handle case of unintentional pypdf `UnicodeDecodeError`
+            # because odoo enforces `strict` mode
+            # https://github.com/versada/odoo/blob/d83bd4f/odoo/tools/pdf.py#L27
+            # and `pypdf` validation may can raise the error when generating message
+            # https://github.com/py-pdf/pypdf/blob/2.x/PyPDF2/_reader.py#L1455
+            stream = BytesIO(watermark)
+            stream.seek(0, os.SEEK_SET)
+            header_byte = stream.read(5)
+            if header_byte != b"%PDF-":
+                # the file is not PDF -- probably this is image (we'll check bellow)
+                pdf_watermark = None
+            else:
+                # if file is PDF -- show the error (no idea where it may come from)
+                raise e
```

Comment:
Should be enough to just re-raise like:
```suggestion
                raise
```